File: apps/minecraft/velocity/src/main/java/crabcraft/net/crabUtilities/velocity/awards/AwardDefinition.java
package crabcraft.net.crabUtilities.velocity.awards;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class AwardDefinition {
    public String id;
    public Reader reader;

    public static class Reader {
        @SerializedName("$type")
        public String type;

        /**
         * Path inside the reader's source object: vanilla stats for standard
         * readers, plugin-provided metrics for {@code custom-int}, or the full
         * stats payload for {@code set-count}.
         */
        public List<String> path;

        /** For match-sum readers: regex patterns whose matching keys are summed. */
        public List<String> patterns;
    }
}

File: apps/minecraft/velocity/src/main/kotlin/crabcraft/net/crabUtilities/velocity/awards/AwardDefinition.kt
package crabcraft.net.crabUtilities.velocity.awards

import com.google.gson.annotations.SerializedName

open class AwardDefinition {
    @JvmField var id: String? = null
    @JvmField var reader: Reader? = null

    open class Reader {
        @SerializedName("\$type") @JvmField var type: String? = null
        /** Path within the reader's vanilla stats, custom metrics or full payload. */
        @JvmField var path: List<String>? = null
        /** Regex patterns for match-sum readers. */
        @JvmField var patterns: List<String>? = null
    }
}
